src/core/server/models/story/story.ts

```diff
@@ -727,3 +742,306 @@ export async function retrieveStorySections(
   // the null entries, the resulting array can not contain null.
   return results.filter((section) => section !== null).sort() as string[];
 }
+
```

Comment:
I would strongly suggest extracting this to a service function, and exposing the MongoDB methods as methods that are imported from each model module. Reaching between collections in a single model file should be avoided at all costs.